### ISSUE TITLE
How can I access the plugin from init.gradle?

### ISSUE THREAD

**Issue by drdtgh** [stevesaliman/gradle-cobertura-plugin#109]
I am trying to write a global `init.gradle` that all of my team's projects can use to get access to the standard reports (codenarc, cobertura) without having to manually update each project.   I am following [the documentation for init.gradle scripts](https://docs.gradle.org/current/userguide/init_scripts.html), and it has worked so far, but not for the cobertura plugin:

```
initscript {
    repositories {
        maven { url "https://plugins.gradle.org/m2/" }
        mavenCentral()
    }
    dependencies {
        classpath "net.saliman:gradle-cobertura-plugin:2.3.1"
    }
}
apply plugin: 'net.saliman.cobertura'
```

```
> gradlew.bat -I config\.gradle\init.gradle tasks
FAILURE: Build failed with an exception.
* Where:
Initialization script 'config\.gradle\init.gradle' line: 49
* What went wrong:
Plugin with id 'net.saliman.cobertura' not found.
```

I think the issue is with the classpath setting, since none of my other reports require one.  Can you advise what I need to do for this to work?  I also tried putting the settings in a `buildscript` block, no joy.

Thank you!

**Comment by drdtgh** [comment]
Oh, yes.  I am using Gradle 2.13, accessed via the gradle wrapper.

**Comment by stevesaliman** [comment]
I did a little digging on this issue, mainly [here](https://stackoverflow.com/questions/14595606/cannot-apply-an-external-plugin-in-a-gradle-init-script-github-project-included/14600210#14600210) and [here](https://discuss.gradle.org/t/apply-plugin-java-fails-with-org-gradle-invocation-defaultgradle-decorated-cannot-be-cast-to-org-gradle-api-internal-project-projectinternal/17430), and I've come up with the following:

- Plugins can't be applied by id.  The implementation class must be used instead.  This is a 3 year old bug, and it doesn't appear to have been fixed yet.
- Plugins shouldn't be applied in an init.gradle script, which is meant for common configuration items
- If you want to apply a plugin from init.gradle, you'll need to use the class name instead of the artifact id.
- If the plugin doesn't support being appled against a Gradle instance, you'll need to put it in an ```allprojects``` block.

I'll look into adding support for applying the cobertura plugin to a Gradle instance, but in the meantime, I think the following workround will work for you:
```groovy
initscript {
    repositories {
        maven { url "https://plugins.gradle.org/m2/" }
        mavenCentral()
    }
    dependencies {
        classpath "net.saliman:gradle-cobertura-plugin:2.3.1"
    }
}
allprojects {
   apply plugin: net.saliman.gradle.plugin.cobertura.CoberturaPlugin
}
```
